lua/various-textobjs/charwise-textobjs.lua

```diff
@@ -351,6 +351,32 @@ function M.mdlink(scope, lookForwL)
 	selectTextobj(pattern, scope, lookForwL)
 end
 
+---@param scope "inner"|"outer" inner selector only includes the content, outer selector includes the type.
+---@param lookForwL integer
+function M.mdEmphasis(scope, lookForwL)
+	local patterns = {}
+	for _, tag in ipairs({ "*", "**", "***", "_", "__", "___" }) do
```

Comment:
Good to consider bold+italics as well. However, there is the edge case of mixed syntax, like someone using `_` for italics and `*` for bold, which results for example in `**_foobar_**`, `_**foobar**_` and so on.

I think rather than complicating it, I'd restrict this textobj to italics OR bold, but not both at once?

Reply:
Good catch. What about just including the simple case here? I googled around, and seems like people tend to first show the case where the characters are all the same. <https://stackoverflow.com/editing-help#italics-bold> only includes that case.

It's a good tradeoff between not including it and over complicating things.

Reply:
I dunno, I feel like `***foobar***` working differently from `**_foobar_**` would feel quite inconsistent to me. And behaving predictably is imo a must for text objects.

Considering that there are I think 6 cases total to consider, factoring bold+italics would make the code more complex than it is worth it for the rather rare edge case of overlapping bold and italics.

Reply:
This was actually a good thing to try code up. That actually helped me fix a bug, where I didn't consider both `_` and `*` in the boundaries.

The code is now stricter and works in all cases. There's basically just 2 extra lines, being that of considering the tag on the right.